modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointManager.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * <p>Represents only an intermediate step in refactoring of checkpointing component and may change in the future.
+ *
+ * <p>This checkpoint ensures that all pages marked as dirty under {@link #checkpointTimeoutLock} will be consistently saved to disk.
+ *
+ * <p>Configuration of this checkpoint allows the following:
+ * <ul>
+ *     <li>Collecting all pages from configured dataRegions which was marked as dirty under {@link #checkpointTimeoutLock}.</li>
+ *     <li>Marking the start of checkpoint on disk.</li>
+ *     <li>Notifying the subscribers of different checkpoint states through {@link CheckpointListener}.</li>
+ *     <li>Synchronizing collected pages with disk using {@link FilePageStoreManager}.</li>
+ * </ul>
+ */
+public class CheckpointManager implements IgniteComponent {
+    /** Checkpoint worker. */
+    private final Checkpointer checkpointer;
+
+    /** Main checkpoint steps. */
+    private final CheckpointWorkflow checkpointWorkflow;
+
+    /** Checkpoint markers storage which mark the start and end of each checkpoint. */
+    private final CheckpointMarkersStorage checkpointMarkersStorage;
+
+    /** Timeout checkpoint lock which should be used while write to memory happened. */
+    private final CheckpointTimeoutLock checkpointTimeoutLock;
+
+    /** Checkpoint page writer factory. */
+    private final CheckpointPagesWriterFactory checkpointPagesWriterFactory;
+
+    /**
+     * Constructor.
+     *
+     * @param logger Logger producer.
+     * @param igniteInstanceName Ignite instance name.
+     * @param checkpointConfig Checkpoint configuration.
+     * @param workerListener Listener for life-cycle checkpoint worker events.
+     * @param longJvmPauseDetector Long JVM pause detector.
+     * @param filePageStoreManager File page store manager.
+     * @param dataRegions Data regions.
+     * @param storagePath Storage path.
+     * @param pageSize Page size in bytes.
+     * @throws IgniteInternalCheckedException If failed.
+     */
+    public CheckpointManager(
+            Function<Class<?>, IgniteLogger> logger,
+            String igniteInstanceName,
+            @Nullable IgniteWorkerListener workerListener,
+            @Nullable LongJvmPauseDetector longJvmPauseDetector,
+            PageMemoryCheckpointConfiguration checkpointConfig,
+            FilePageStoreManager filePageStoreManager,
+            Collection<PageMemoryDataRegion> dataRegions,
+            Path storagePath,
+            // TODO: IGNITE-17017 Move to common config
+            int pageSize
+    ) throws IgniteInternalCheckedException {
+        PageMemoryCheckpointView checkpointConfigView = checkpointConfig.value();
+
+        ReentrantReadWriteLockWithTracking reentrantReadWriteLockWithTracking = checkpointConfigView.logReadLockHolders()
+                ? new ReentrantReadWriteLockWithTracking(logger.apply(CheckpointReadWriteLock.class), 5_000)
```

Comment:
What's 5000?

Reply:
5_000 mills == 5 sec

Expanded comments for **ReentrantReadWriteLockWithTracking**.

Reply:
**PageMemoryCheckpointConfigurationSchema#logReadLockThresholdTimeout with default 0** Will be better?

Reply:
Yes, I guess so

Reply:
Fix it.